slick/src/main/scala/slick/jdbc/HsqldbProfile.scala

```diff
@@ -118,8 +121,169 @@ trait HsqldbProfile extends JdbcProfile {
     }
     override val uuidJdbcType = new UUIDJdbcType {
       override def sqlType = java.sql.Types.BINARY
+
       override def sqlTypeName(sym: Option[FieldSymbol]) = "BINARY(16)"
     }
+    override val localTimeType = new LocalTimeJdbcType {
+      override def sqlType = java.sql.Types.TIME
+
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIME(3)"
+
+      override def getValue(r: ResultSet, idx: Int): LocalTime = {
+        // Fixes an issue caused because Hsqldb outputs the time in a non-standard format
+        // not adding trailing zeros to the hours. For example: '2:14:41.421' instead of '02:14:41.421'
+        r.getString(idx) match {
+          case null => null
+          case isoTime if isoTime.indexOf(':') == 2 => LocalTime.parse(isoTime)
+          case isoTime => LocalTime.parse(s"0$isoTime")
```

Comment:
not urgent. This is a big PR, will take me a while to get through it. Maybe some other people want to jump in and take a look :). @tminglei @nafg @szeiger . One concern to keep in mind (which probably has been addressed already) is that Slick should be DB centric in the sense that the type of the value in the DB matters and should be sensible. The type mapping is just a decoding into a reasonable scala type, but the DB type is the central one. That's how I see it.